CHANGELOG.md

```diff
@@ -1,5 +1,20 @@
 # Changelog
 
+## 0.17.0 - 2024-07-20
+
+### Features
+
+* Add a new rule `DEP005` to detect project dependencies that are in the standard library. ([#761](https://github.com/fpgmaas/deptry/pull/761))
+
+### Miscellaneous
+
+* Various improvements in the CI pipeline ([#755](https://github.com/fpgmaas/deptry/pull/755) and ([#767](https://github.com/fpgmaas/deptry/pull/767)))
```

Comment:
Since this is not a user-facing change and only relevant to us, IMO it's not necessary to mention this.